ste/sender-blockBlob.go

```diff
@@ -273,6 +273,32 @@ func (s *blockBlobSenderBase) Cleanup() {
 	}
 }
 
+//Currently we've common Metadata Copier across all senders for block blob.
+func (s *blockBlobSenderBase) GenerateCopyMetadata(id common.ChunkID) chunkFunc {
+	return createChunkFunc(true, s.jptm, id, func() {
+		if s.jptm.Info().PreservePOSIXProperties {
+
+			if unixSIP, ok := s.sip.(IUNIXPropertyBearingSourceInfoProvider); ok {
+				// Clone the metadata before we write to it, we shouldn't be writing to the same metadata as every other blob.
+				s.metadataToApply = common.Metadata(s.metadataToApply).Clone().ToAzBlobMetadata()
+	
+				statAdapter, err := unixSIP.GetUNIXProperties()
+				if err != nil {
+					s.jptm.FailActiveSend("GetUNIXProperties", err)
```

Comment:
so if we fail to get the properties, we fail entire transfer ? is it what customer would expect ?

Reply:
Yes, it is a hard failure generally in all our scenarios